encryption.md

```diff
@@ -0,0 +1,48 @@
+## Message Encryption
+
+DIDComm supports two types of message encryption: Authenticated Sender Encryption and Anonymous Sender Encryption. Both forms are encrypted to the recipient, but only Authenticated Sender Encryption provides assurances of who the sender is. 
+
+The encrypted form of a JWM is a JWE. The JOSE family defines [JSON Web Algorithms](https://tools.ietf.org/html/rfc7518) (JWAs) which standardize certain cryptographic operations that are related to preparing JOSE structures. For the purposes of interoperability, DIDComm messaging does not support all JWAs, rather it takes a subset of the supported algorithms that are applicable for the following cases around secure messaging. These supported algorithms are listed here.
+
+
+### Sender Authenticated Encryption
+
+When a sender would like to encrypt a message for a recipient or multiple recipients and also be authenticated such that the recipients can determine they were the party that encrypted the message, the JWA of `ECDH-1PU` defined by [draft](https://tools.ietf.org/html/draft-madden-jose-ecdh-1pu-02) SHOULD be used within the structure of a JWE. 
+
+For the keys involved in key agreement, the following elliptic curves MUST be supported.
+
+|Curve |Description                                                                                                                        |
+|------|-----------------------------------------------------------------------------------------------------------------------------------|
+|X25519|The underlying curve is actually `Curve25519`, however when used in the context of diffie helman the identifier of `X25519` is used|
+|P-256 |NIST defined P-256 elliptic curve                                                                                                  |
+
+For content encryption of the message, the following algorithms MUST be supported.
+
+|Algorithm(JWA) |Description                                          |
+|---------------|-----------------------------------------------------|
+|C20P           |ChaCha20Poly1305                                     |
+|AES-GCM        |Advanced Encryption Standard with galois/counter mode|
```

Comment:
We can use random nonces for AES-GCM but need to be careful the number of messages sent. Practically speaking we shouldn't hit the limit (2^32 message), but need to state security considerations ( paging @mikelodder7 or @christianlundkvist ) around how nonces should be generated here or reference to an RFC which states how to do this safely (preferable).

Reply:
Of note because I'm sure it will be brought up Ideally, we would just use AES-GCM-SIV, but I don't believe that's NIST registered which is the reason we're supporting AES-GCM. For a nonce resistant cipher implementers should opt for XC20P if they don't have constraints on being NIST compliant. We should probably mention this as well.

Reply:
I've added a todo about nonces. Is that sufficient for this update?